server/datastore/mysql/software_test.go

```diff
@@ -31,40 +31,40 @@ func TestSoftware(t *testing.T) {
-		{"UpsertSoftwareCPEs", testUpsertSoftwareCPEs},
-		{"DeleteOutOfDateVulnerabilities", testDeleteOutOfDateVulnerabilities},
-		{"DeleteSoftwareCPEs", testDeleteSoftwareCPEs},
-		{"SoftwareByIDNoDuplicatedVulns", testSoftwareByIDNoDuplicatedVulns},
-		{"SoftwareByIDIncludesCVEPublishedDate", testSoftwareByIDIncludesCVEPublishedDate},
-		{"getHostSoftwareInstalledPaths", testGetHostSoftwareInstalledPaths},
-		{"hostSoftwareInstalledPathsDelta", testHostSoftwareInstalledPathsDelta},
-		{"deleteHostSoftwareInstalledPaths", testDeleteHostSoftwareInstalledPaths},
-		{"insertHostSoftwareInstalledPaths", testInsertHostSoftwareInstalledPaths},
-		{"VerifySoftwareChecksum", testVerifySoftwareChecksum},
+		// {"SaveHost", testSoftwareSaveHost},
+		// {"CPE", testSoftwareCPE},
+		// {"HostDuplicates", testSoftwareHostDuplicates},
+		// {"LoadVulnerabilities", testSoftwareLoadVulnerabilities},
+		// {"ListSoftwareCPEs", testListSoftwareCPEs},
+		// {"NothingChanged", testSoftwareNothingChanged},
+		// {"LoadSupportsTonsOfCVEs", testSoftwareLoadSupportsTonsOfCVEs},
+		// {"List", testSoftwareList},
+		// {"SyncHostsSoftware", testSoftwareSyncHostsSoftware},
+		// {"DeleteSoftwareVulnerabilities", testDeleteSoftwareVulnerabilities},
+		// {"HostsByCVE", testHostsByCVE},
+		// {"HostVulnSummariesBySoftwareIDs", testHostVulnSummariesBySoftwareIDs},
+		// {"UpdateHostSoftware", testUpdateHostSoftware},
+		// {"UpdateHostSoftwareDeadlock", testUpdateHostSoftwareDeadlock},
+		// {"UpdateHostSoftwareUpdatesSoftware", testUpdateHostSoftwareUpdatesSoftware},
+		// {"ListSoftwareByHostIDShort", testListSoftwareByHostIDShort},
+		// {"ListSoftwareVulnerabilitiesByHostIDsSource", testListSoftwareVulnerabilitiesByHostIDsSource},
+		// {"InsertSoftwareVulnerability", testInsertSoftwareVulnerability},
+		// {"ListCVEs", testListCVEs},
+		// {"ListSoftwareForVulnDetection", testListSoftwareForVulnDetection},
+		// {"AllSoftwareIterator", testAllSoftwareIterator},
+		// {"AllSoftwareIteratorForCustomLinuxImages", testSoftwareIteratorForLinuxKernelCustomImages},
+		// {"UpsertSoftwareCPEs", testUpsertSoftwareCPEs},
+		// {"DeleteOutOfDateVulnerabilities", testDeleteOutOfDateVulnerabilities},
+		// {"DeleteSoftwareCPEs", testDeleteSoftwareCPEs},
+		// {"SoftwareByIDNoDuplicatedVulns", testSoftwareByIDNoDuplicatedVulns},
+		// {"SoftwareByIDIncludesCVEPublishedDate", testSoftwareByIDIncludesCVEPublishedDate},
+		// {"getHostSoftwareInstalledPaths", testGetHostSoftwareInstalledPaths},
+		// {"hostSoftwareInstalledPathsDelta", testHostSoftwareInstalledPathsDelta},
+		// {"deleteHostSoftwareInstalledPaths", testDeleteHostSoftwareInstalledPaths},
+		// {"insertHostSoftwareInstalledPaths", testInsertHostSoftwareInstalledPaths},
+		// {"VerifySoftwareChecksum", testVerifySoftwareChecksum},
 		{"ListHostSoftware", testListHostSoftware},
-		{"SetHostSoftwareInstallResult", testSetHostSoftwareInstallResult},
+		// {"SetHostSoftwareInstallResult", testSetHostSoftwareInstallResult},
```

Comment:
oops! seems like all of this is not intentional?

I use the `-run=` flag in `go test` for this, for this case it would be `-run=TestSoftware/ListHostSoftware`

I don't know if I already shared this with you, but given the amount of flags for tests we have, I have a shell function just for running tests:

```sh
function pkg-test() {
  # NETWORK_TEST=1 disabled
  # MINIO_STORAGE_TEST=1 SAML_IDP_TEST=1
  MYSQL_TEST=1 REDIS_TEST=1 SAML_IDP_TEST=1 go test github.com/fleetdm/fleet/v4/$1 -v -run=$2 -race=true -parallel=8 -coverprofile=coverage.txt -covermode=atomic -failfast -tags full
}
```

In this case, I would run:

```
$ pkg-test server/datastore/mysql TestSoftware/ListHostSoftware
```

Reply:
Ah yeah I forget to remove these comments all the time, I do it because I run the tests from the VSCode UI and it has problems with running specific subtests. That's a handy function tho, it'll help me out a lot!